deltachat-ffi/deltachat.h

```diff
@@ -2630,6 +2636,116 @@ char* dc_get_last_error (dc_context_t* context);
 void dc_str_unref (char* str);
 
 
+/**
+ * @class dc_backup_provider_t
+ *
+ * Set up another device.
+ */
+
+/**
+ * Creates an object for sending a backup to another device.
+ *
+ * The backup is sent to through a peer-to-peer channel which is bootstrapped
+ * by a QR-code.  The backup contains the entire state of the account
+ * including credentials.  This can be used to setup a new device.
+ *
+ * Once this function returns, the backup is being offered to remote devices.
+ * To wait until one device received the backup, use
+ * dc_backup_provider_wait().  Alternatively abort the operation using
+ * dc_stop_ongoing_process().
+ *
+ * During execution of the job #DC_EVENT_IMEX_PROGRESS is sent out to indicate
+ * state and progress.
+ *
+ * @memberof dc_backup_sender_t
+ * @param context The context.
+ * @return Opaque object for sending the backup.
+ *    On errors, NULL is returned and dc_get_last_error() returns an error that
+ *    should be shown to the user.
+ */
+dc_backup_provider_t* dc_backup_provider_new (dc_context_t* context);
```

Comment:
`dc_backup_provider_new()` calls `BackupProvider::prepare()` which says that all  I/O must be stopped.

i assume, this hint is missing here as well, similar to dc_configure, we should say: "During sending backups, IO must not be started, if needed stop IO using dc_accounts_stop_io() or dc_stop_io() first."

it is not clear to me, when IO can be restarted - already after `dc_backup_provider_new()` returns - or after `dc_backup_provider_wait()` is done?

wrt blocking: `dc_backup_provider_new()` as well as `dc_backup_provider_wait()` are blocking,  may take minutes and should be called in the UI from within a thread, right? (to allow canceling, progress bar etc) i'd add a line to clarify this as well.

Reply:
> `dc_backup_provider_new()` calls `BackupProvider::prepare()` which says that all I/O must be stopped.
> 
> i assume, this hint is missing here as well, similar to dc_configure, we should say: "During sending backups, IO must not be started, if needed stop IO using dc_accounts_stop_io() or dc_stop_io() first."

Yep, will updated.

> it is not clear to me, when IO can be restarted - already after `dc_backup_provider_new()` returns - or after `dc_backup_provider_wait()` is done?

io can only be restarted after dc_backup_provider_wait has returned.

> wrt blocking: `dc_backup_provider_new()` as well as `dc_backup_provider_wait()` are blocking, may take minutes and should be called in the UI from within a thread, right? (to allow canceling, progress bar etc) i'd add a line to clarify this as well.

I was only considering `_wait()` to be blocking, but now you mention it during `_new()` we also already export the database so yeah, that's blocking as well.  Once it returns the provider is ready and a remote can connect.

Reply:
in theory you could avoid a thread for the `dc_backup_provider_wait` if you first wait for the IMEX event 1000 or 0 and only then call `_wait()` as then it wouldn't block.  Not sure if that's interesting for you or not.

Reply:
thanks for fast response and clarifying!